tools/reproduce_comparison/Jenkinsfile

```diff
@@ -14,27 +14,34 @@ pipeline {
     stages {
         stage('Lets Compare') {
             agent {
-                label "${params.platform}"
-            }
+                label "${params.platform}&&build"
+            } 
             steps {
                 script {
-                    cleanWs()
                     try {
                         echo "Fetching artifact1 from ${params.URL1}"
-                        sh "mkdir url1Dir1 && wget -q ${params.URL1} -O - | tar -xz -C url1Dir1"
+                        def ret1 = sh returnStatus: true, script: "mkdir url1Dir1 && wget -q ${params.URL1} -O - | tar -xz -C url1Dir1"
                         echo "Fetching artifact2 from ${params.URL2}"
-                        sh "mkdir url1Dir2 && wget -q ${params.URL2} -O - | tar -xz -C url1Dir2"
+                        def ret2 = sh returnStatus: true, script: "mkdir url1Dir2 && wget -q ${params.URL2} -O - | tar -xz -C url1Dir2"
+                        if (ret1 || ret2) { 
+                            currentBuild.result = "UNSTABLE"
+                            error "Stopping after uncompress tar file"
+                        }
+
                         // call extra platform specific function
                         if ( params.platform != 'linux' ) { 
                             "prep${platform}" ("url1Dir1", "url1Dir2") 
                         }
-                        retVal = sh returnStatus: true, script: "diff -q -r url1Dir1 url1Dir2"
+                        def retVal = sh returnStatus: true, script: "diff -q -r url1Dir1 url1Dir2"
                         if (retVal != 0 ){
-                            echo "Error: two builds are not the same!"
                             currentBuild.result = 'FAILURE'
+                            error "Error: two builds are not the same!"
```

Comment:
Might be worth adding the diff output

Reply:
from test https://ci.adoptopenjdk.net/view/work-in-progress/job/wen_reproducible_comparision/9/console
it lists all the files before prints `Error: two builds are not the same!`